src/kinetics/ReactionFactory.cpp

```diff
@@ -0,0 +1,242 @@
+    addAlias("chemically-activated", "chemically_activated");
+    reg_XML("chemically-activated",
+            [](Reaction* R, const XML_Node& node) {
+                setupChemicallyActivatedReaction(*(ChemicallyActivatedReaction*)R, node);
+            });
+    reg_AnyMap("chemically-activated",
+               [](Reaction* R, const AnyMap& node, const Kinetics& kin) {
+                   setupFalloffReaction(*(FalloffReaction*)R, node, kin);
+               });
+
+    // register pressure-depdendent-Arrhenius reactions
+    reg("pressure-dependent-Arrhenius", []() { return new PlogReaction(); });
+    addAlias("pressure-dependent-Arrhenius", "plog");
+    addAlias("pressure-dependent-Arrhenius", "pdep_arrhenius");
+    reg_XML("pressure-dependent-Arrhenius",
+            [](Reaction* R, const XML_Node& node) {
+                setupPlogReaction(*(PlogReaction*)R, node);
+            });
+    reg_AnyMap("pressure-dependent-Arrhenius",
+               [](Reaction* R, const AnyMap& node, const Kinetics& kin) {
+                   setupPlogReaction(*(PlogReaction*)R, node, kin);
+               });
+
+    // register Chebyshev reactions
+    reg("Chebyshev", []() { return new ChebyshevReaction(); });
+    addAlias("Chebyshev", "chebyshev");
+    reg_XML("Chebyshev",
+            [](Reaction* R, const XML_Node& node) {
+                setupChebyshevReaction(*(ChebyshevReaction*)R, node);
+            });
+    reg_AnyMap("Chebyshev",
+               [](Reaction* R, const AnyMap& node, const Kinetics& kin) {
+                   setupChebyshevReaction(*(ChebyshevReaction*)R, node, kin);
+               });
+
+    // register custom reactions specified by Func1 objects
+    reg("custom-rate-function", []() { return new CustomFunc1Reaction(); });
+    reg_XML("custom-rate-function",
+            [](Reaction* R, const XML_Node& node) {
+                throw CanteraError("ReactionFactory", "Custom reactions based "
+                    "on 'Func1' objects cannot be created from XML nodes'");
+            });
+    reg_AnyMap("custom-rate-function",
+               [](Reaction* R, const AnyMap& node, const Kinetics& kin) {
+                   R->setup(node, kin);
+               });
+
+    // register custom Python reactions
+    reg("elementary-new", []() { return new TestReaction(); });
+    reg_XML("elementary-new",
+            [](Reaction* R, const XML_Node& node) {
+                throw CanteraError("ReactionFactory", "Test reactions "
+                                   "cannot be created from XML nodes'");
+            });
+    reg_AnyMap("elementary-new",
+               [](Reaction* R, const AnyMap& node, const Kinetics& kin) {
+                   R->setup(node, kin);
+               });
+
+    // register interface reactions
+    reg("interface", []() { return new InterfaceReaction(); });
+    addAlias("interface", "surface");
+    addAlias("interface", "edge");
+    addAlias("interface", "global");
```

Comment:
I think `global` is another of those unimplemented vestigial rate types.

Reply:
Since this is based on existing code, I am opting for:
```
addDeprecatedAlias("interface", "global");
```

Reply:
Nevermind ... couldn't locate it. Deleting ...